src/Service.GraphQLBuilder/GraphQLStoredProcedureBuilder.cs

```diff
@@ -37,24 +38,30 @@ public static class GraphQLStoredProcedureBuilder
             // which are needed because parameter and column names can differ.
             StoredProcedureDefinition spdef = (StoredProcedureDefinition)dbObject.SourceDefinition;
 
-            // Create input value definitions from parameters defined in runtime config.
-            if (entity.Source.Parameters is not null)
+            // Create input value definitions from parameters defined in stored proc.
+            if (spdef is not null)
             {
-                foreach (string param in entity.Source.Parameters.Keys)
+                foreach ((string param, ParameterDefinition definition) in spdef.Parameters)
                 {
                     // Input parameters defined in the runtime config may denote values that may not cast
                     // to the exact value type defined in the database schema.
                     // e.g. Runtime config parameter value set as 1, while database schema denotes value type decimal.
                     // Without database metadata, there is no way to know to cast 1 to a decimal versus an integer.
-                    string defaultValueFromConfig = entity.Source.Parameters[param].ToString()!;
-                    Tuple<string, IValueNode> defaultGraphQLValue = ConvertValueToGraphQLType(defaultValueFromConfig, parameterDefinition: spdef.Parameters[param]);
+
+                    IValueNode? valueNode = null;
+                    if (entity.Source.Parameters is not null && entity.Source.Parameters.TryGetValue(param, out object? value))
```

Comment:
Instead of retrieving the default value, is it not sufficient to just have the metadata which tells whether there is a default value available for a stored proc param? In that case, we can just invoke the SP without passing in a value for that param in case no value was specified by the user.

Reply:
Ive added description in the pr header on why this is very hard to do. The existing solutions are flaky and it requires proper review. However, this pr main aim is to unblock you from having to readd every parameter from your sql db into your config even without lack of default values. Now atleast customer can respecify the default value in the config or have it in their frontend code and atleast make queries without having to repeat every parameter in the config.